app/jobs/retention_policy_job.rb

```diff
@@ -0,0 +1,31 @@
+class RetentionPolicyJob < ApplicationJob
+  queue_as :low
+
+  def scopes
+    Enumerator.new do |y|
+      JobApplication.after_submission.where(submitted_at: ...threshold).tap do |job_applications|
+        # self-disclosure data
+        y << SelfDisclosure.joins(self_disclosure_request: :job_application).merge(job_applications)
+        y << SelfDisclosureRequest.joins(:job_application).merge(job_applications)
+        # reference data
+        y << JobReference.joins(referee: :job_application).merge(job_applications)
+        y << ReferenceRequest.joins(referee: :job_application).merge(job_applications)
+        # qualifying active job applications
+        y << job_applications
+      end
+
+      # draft job application when vacancy expired
+      y << JobApplication.joins(:vacancy).draft.where(updated_at: ...threshold).merge(Vacancy.expired)
+
+      y << Feedback.where(created_at: ...threshold) if hard_delete?
+    end
+  end
+
+  def threshold
+    raise "define thresold period in subclass"
+  end
```

Comment:
I'm a bit unsure about using inheritance for this - Josh Bloch says normally to use composition rather than inheritance. Actually Ruby make a potentially neat solution possible, as we have functions as parameters so this could all be in the subclass, passing parameters (including the &:discard or &:delete method). WDYT?